src/platform/android/java/chip/platform/DiagnosticDataProviderImpl.java

```diff
@@ -0,0 +1,107 @@
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+package chip.platform;
+
+import android.content.Context;
+import android.util.Log;
+import java.net.Inet4Address;
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.SocketException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+
+public class DiagnosticDataProviderImpl implements DiagnosticDataProvider {
+  private static final String TAG = DiagnosticDataProviderImpl.class.getSimpleName();
+  private Context mContext;
+
+  public DiagnosticDataProviderImpl(Context context) {
+    mContext = context;
+  }
+
+  @Override
+  public int getRebootCount() {
+    // TODO: boot count is not supported until api 24, please do it at factory app
+    //        try {
+    //            return Settings.Global.getInt(mContext.getContentResolver(),
+    // Settings.Global.BOOT_COUNT);
+    //        } catch (Settings.SettingNotFoundException e) {
+    //            Log.e(TAG, "getRebootCount: " + e.getMessage());
+    //        }
+    return 100;
+  }
+
+  @Override
+  public NetworkInterface[] getNetworkInterfaces() {
+    try {
+      ArrayList<java.net.NetworkInterface> networkInterfaces =
+          Collections.list(java.net.NetworkInterface.getNetworkInterfaces());
+      int size = networkInterfaces.size();
+      List<NetworkInterface> destInterfaces = new ArrayList<>();
+      for (int i = 0; i < size; i++) {
+        java.net.NetworkInterface nif = networkInterfaces.get(i);
+        String name = nif.getName();
+
+        if (name != null && (name.startsWith("eth") || name.startsWith("wlan"))) {
+          NetworkInterface anInterface = new NetworkInterface();
+          anInterface.isOperational = nif.isUp();
+          anInterface.hardwareAddress = nif.getHardwareAddress();
+          anInterface.name = nif.getName();
+          anInterface.type =
+              name.startsWith("wlan")
+                  ? NetworkInterface.INTERFACE_TYPE_WI_FI
+                  : NetworkInterface.INTERFACE_TYPE_ETHERNET;
+
+          anInterface.offPremiseServicesReachableIPv4 = NetworkInterface.REACHABLE_UNKNOWN;
+          anInterface.offPremiseServicesReachableIPv6 = NetworkInterface.REACHABLE_UNKNOWN;
+          for (Enumeration<InetAddress> inetAddresses = nif.getInetAddresses();
+              inetAddresses.hasMoreElements(); ) {
+            InetAddress inetAddress = inetAddresses.nextElement();
+            if (inetAddress instanceof Inet4Address) {
+              anInterface.offPremiseServicesReachableIPv4 = NetworkInterface.REACHABLE_YES;
```

Comment:
This does not look right.  Just because we can speak IPv4 does not mean we can reach whatever server we actually care about over IPv4 right this second.

This generic code has no way to know whether things are reachable or not.  That information needs to be provided by the application.

Reply:
fixed